presto-hive/src/main/java/com/facebook/presto/hive/AbstractDwrfEncryptionInformationSource.java

```diff
@@ -106,7 +106,12 @@ protected abstract EncryptionInformation getWriteEncryptionInformationInternal(
 
         Map<String, String> fieldToKeyReference;
         if (encryptTable.isPresent()) {
-            fieldToKeyReference = ImmutableMap.of(DwrfEncryptionMetadata.TABLE_IDENTIFIER, encryptTable.get());
+            if (!requestedColumns.isPresent() || !requestedColumns.get().isEmpty()) {
```

Comment:
why would requested columns be absent?

Reply:
why would it be absent (requestedColumns.isPresent()) or the underlying set by empty (requestedColumns.get.isEmpty())?

For `requestedColumns.isPresent()` it is being populated by an optimizer and it is easy for a company to not have that optimizer in their list of optimizers and that is when this will be empty. The implication is that we will try to get the keys for all columns and hence the user needs to have access to everything in the table.

For `requestedColumns.get().isEmpty()) - this happens when you have a query like `SELECT partitionCol FROM table` or `SELECT count(*) FROM table` where we are not selecting or filtering on a column.